### DISCUSSION TITLE
Failed to resolve entry for package "danfojs"

### DISCUSSION THREAD

**Discussion by barmatey** [opensource9ja/danfojs#525] [Q&A]
Hi, guys!
I'm trying to use Danfojs+Vite+Vue3 but I have the import problem. 
Screenshot with full text in attachments. 
Please tell me how could I resolve the problem?
Thank you.

![image](https://user-images.githubusercontent.com/40522008/203565725-8237960b-4532-4231-b4cf-1c959c8b2575.png)

**Comment by hopezh** [comment]
I got the same issue after "npm install danfojs" and "npm run dev"...

**Comment by tnobody** [comment]
Hi, I had the same error. I figured out that the file that is referenced in the `module`-property in the `package.json` (`lib/bundle-esm.js`) is somehow not included in the npm package.

Interestingly the files are generated when I ran the following commands:
```bash
git clone https://github.com/javascriptdata/danfojs/
cd danfojs
yarn run install
cd src/danfojs-browser
yarn build
```

The last command invokes all other build commands and generates the expected file.
Unfortunately, I was not able to find any reason why these files didn't make it in the released package on npm.

Since I also didn't find any hint on how danfojs is released I'm not able to help with any kind of PR or something else (for now at least)

_Update:_

I found a workaround for now (also in a vite + typescript setup):

- I'v created a file `types/dfd.d.ts` with following contents:
```typescript
import * as danfojs from "danfojs";

declare global {
  const dfd: typeof danfojs;
}
```
- added this file to the `compilerOptions.types`-array in `tsconfig.json`
- added the cdn version in the `index.html` of my project (`<script src="https://cdn.jsdelivr.net/npm/danfojs@1.1.2/lib/bundle.min.js"></script>`)

This also has the little advantage that the CDN version can be loaded very fast since browser will manage the caching (danfojs browser bundle is roughly ~6MB)

**Reply by S-L-Moore** [reply]
Thank you, I doubt I would've got this running without your helpful post.

For anyone else who stumbles over this, I placed the dfd.d.ts under ./src/lib/types without issue & added the danfojs to the compilerOptions.types array as follows:
```
{
  "extends": "@tsconfig/svelte/tsconfig.json",
  "compilerOptions": {
    "types": ["danfojs"],
    ...
  },
  "include": ["src/**/*"],
  "references": [{ "path": "./tsconfig.node.json" }]
}
```

**Reply by Sambuxc** [reply]
Thanks! Is this code going in the Vite.config.js file?

**Reply by hopezh** [reply]
Hi @S-L-Moore and @tnobody, can you kindly explain a bit more on how your work-around is done in detail? I've tried, **very naively**, to add the `dfd.d.ts` and `tsconfig.json` files as shown below. But I still can't get this issue solved. (I'm initing a `javascript` project using `vite`). See the code [here](https://github.com/hopezh/test_plotlyjs_map/blob/main/src/main.js). 

![image](https://github.com/javascriptdata/danfojs/assets/23435810/57355efe-235d-4c3b-9417-92de4a4046d1)

**Reply by lassebomh** [reply]
### Install danfojs

`npm i danfojs`

### Create the `dfd.d.ts` file

```typescript
// src/lib/types/dfd.d.ts

import * as danfojs from "danfojs";

declare global {
  const dfd: typeof danfojs;
}
```
### Add it to tsconfig.json

> added this file to the compilerOptions.types-array in tsconfig.json

This means that you need to add the path of the `dfd.d.ts` file to the `compilerOptions.types` option inside `tsconfig.json`.

Ingoring all other options, the file would look like this:

```jsonc
{
  // ...
  "compilerOptions": {
    // ...
    "types": ["./src/lib/types/dfd.d.ts"]
    // ...
  }
  // ...
}
```

My tsconfig looked like this afterwards:

```jsonc
// tsconfig.json
{
  "extends": "@tsconfig/svelte/tsconfig.json",
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "module": "ESNext",
    "resolveJsonModule": true,
    "allowJs": true,
    "checkJs": true,
    "types": ["./src/lib/types/dfd.d.ts"], // I added this line
    "isolatedModules": true
  },
  "include": ["src/**/*.d.ts", "src/**/*.ts", "src/**/*.js", "src/**/*.svelte"],
  "references": [{ "path": "./tsconfig.node.json" }]
}
```
### Add the DanfoJS bundle to your `index.html`
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/vite.svg" />

    <!-- I added this line -->
    <script src="https://cdn.jsdelivr.net/npm/danfojs@1.1.2/lib/bundle.js"></script>
    
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite + Svelte + TS</title>
  </head>
  <body>
    <div id="app"></div>
    <script type="module" src="/src/main.ts"></script>
  </body>
</html>
```

Make sure this version of DanfoJS matches the one you installed through npm and that it runs **before** the `main.ts` script.

**Reply by hopezh** [reply]
@lassebomh thanks a lot. I'll give it a try. I'm working on a JavaScript project, not TypeScript. Will this be a problem, considering that the config file is for typescript specifically?

**Comment by benjamin-white** [comment]
I also had this issue and though not ideal, downgrading from `"danfojs": "^1.1.2"` to `"danfojs": "1.1.1"` in `package.json` fixed it for me. Hopefully it's resolved in a future patch or the solution above might be a better idea, but just in case it's useful downgrading is another option.

**Reply by sewonjun** [reply]
Thanks!! It works on Vite!! :)

**Comment by risenW** [comment]
Fixed in latest release 1.2.0

**Reply by Steffen278** [reply]
I think I am still getting this error.
Used npm install danfojs, parcel src/index.html
![iScreen Shoter - Google Chrome - 250609133327](https://github.com/user-attachments/assets/e3cc7a0a-1e37-488e-af40-0fe9a266d2c2)
